### PR TITLE
Experimental player textures & logs!

### PR DESCRIPTION
**Pull request by Liam0**
Added logging to all admin actions.

Logging is using a Database function instead of PublicVariableEventHandler(it can still print to rpt if u have any reason to do so)

Changed the getID button in the admin menu to display the log dialog showing admin and player logs, initial commits actually had this in the commit messages but i removed those when i squashed the commits together just to make it nice and clean for anyone who has to read/implement it. it should now be displayed alongside the rest of the playerinfo within the admin menu.

Is there even anyone using getID??

Player textures converted to use a config file ie.
{ "textures\cop_uniform.jpg", "life_coplevel", "SCALAR", 1, "cop"},

Additional fixes

Typo in localize string, fn_restrain.sqf
Fix to stop vinteraction menu from opening while restrained.
Fix to stop players from entering turrets on vehicles while being restrained.

### CODE REVIEW

File: Altis_Life.Altis/Config_Master.hpp [hunk 1]
```diff
@@ -4,182 +4,184 @@
...
-        {"STR_Crime_20","500","20"},
-        {"STR_Crime_21","500","21"},
-        {"STR_Crime_22","2000","22"},
-        {"STR_Crime_23","5000","23"},
-        {"STR_Crime_24","10000","24"},
-        {"STR_Crime_25","20000","25"}
-    };
+/* crimes[] = {String, Bounty, Code} */
+crimes[] = {
+		{"STR_Crime_187V","650","187V"},
```
Comment (sargken) [inline Altis_Life.Altis/Config_Master.hpp]: should only be 1 tab for indentation but that's just my opinion.

File: Altis_Life.Altis/Config_Master.hpp [hunk 2]
```diff
@@ -4,182 +4,184 @@
...
+	vehicle_rental_multiplier_COP = .3; //Cop Vehicle Rental Price = Config_Vehicle price * multiplier
+	vehicle_rental_multiplier_MEDIC = .55; //Medic Vehicle Rental Price = Config_Vehicle price * multiplier
+	vehicle_rental_multiplier_OPFOR = -1; // -- NOT IN USE -- Simply left in for east support.
+	
+	/* Vehicle Sell Prices */
+	vehicle_sell_multiplier_CIVILIAN = .5; //Civilian Vehicle Garage Sell Price = Vehicle Buy Price * multiplier
+	vehicle_sell_multiplier_COP = .5; //Cop Vehicle Garage Sell Price = Vehicle Buy Price * multiplier
+	vehicle_sell_multiplier_MEDIC = .5; //Medic Vehicle Garage Sell Price = Vehicle Buy Price * multiplier
+	vehicle_sell_multiplier_OPFOR = -1; // -- NOT IN USE -- Simply left in for east support.
+
```
Comment (sargken) [inline Altis_Life.Altis/Config_Master.hpp]: Should not leave tabs in empty lines.

**Comment by Liam0** [comment]
@sargken there u go.

**Comment by Liam0** [comment]
i didn't localize because i know @Jawshy is working on localisation right now and i dont want to screw him over with conflicts.

**Comment by tkcjesse** [comment]
Won't logging this way increase server overhead substantially?

**Comment by Liam0** [comment]
@tkcjesse, yeah i'll just remove that, ur right it shouldn't be necessary. And that depends on how much stuff u want to log i guess. Hopefully it doesn't.

**Comment by tkcjesse** [comment]
Also, Why'd you remove getID?

**Comment by Liam0** [comment]
it's still there..

File: Altis_Life.Altis/core/admin/fn_admingetID.sqf [hunk 3]
```diff
@@ -1,16 +0,0 @@
-/*
-	File: fn_admingetID.sqf
```
Comment (tkcjesse) [inline Altis_Life.Altis/core/admin/fn_admingetID.sqf]: @Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

File: Altis_Life.Altis/Functions.hpp [hunk 4]
```diff
@@ -68,16 +68,15 @@ class Life_Client_Core {
 		class adminCompensate {};
 		class adminDebugCon {};
 		class adminFreeze {};
-		class admingetID {};
```
Comment (tkcjesse) [inline Altis_Life.Altis/Functions.hpp]: @Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

File: Altis_Life.Altis/CfgRemoteExec.hpp [hunk 5]
```diff
@@ -88,10 +89,10 @@ class CfgRemoteExec {
         F(TON_fnc_chopShopSell,SERVER)
         F(TON_fnc_cleanupRequest,SERVER)
         F(TON_fnc_deleteDBContainer,SERVER)
-        F(TON_fnc_getID,SERVER)
```
Comment (tkcjesse) [inline Altis_Life.Altis/CfgRemoteExec.hpp]: @Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

File: Altis_Life.Altis/dialog/admin_menu.hpp [hunk 6]
```diff
@@ -39,8 +39,8 @@ class life_admin_menu {
 
 		class RscButtonMenu_2402: Life_RscButtonMenu {
 			idc = -1;
-			text = "$STR_Admin_GetID";
```
Comment (tkcjesse) [inline Altis_Life.Altis/dialog/admin_menu.hpp]: @Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

**Comment by tkcjesse** [comment]
@Liam0 Help me out, I see where you removed & wrote over getID, but where is it now? I don't see it. Help me out please.

**Comment by Liam0** [comment]
@tkcjesse fn_admininfo.sqf

**Comment by tkcjesse** [comment]
Did I try it? No. Can't be bothered reading every single change in 49 files... I read it yes, briefly. Considering I obviously noticed it was removed and changed. Sorry I didn't notice 2 extremely small changes to a single file that has the changes to where I need to scroll to the right. In 49 changed files. HENCE why I asked you to please help me out. It wasn't meant rude or anything. Just didn't see it anywhere, considering your "ChangeLog" isn't exactly detailed to explain what you did with it. Hell, it's not even mentioned.

**Comment by Liam0** [comment]
Im not asking you to read every single addition but maybe read the relevant ones if there is something u dont understand. i can't spend more time on the changelog then the actual files

But here you go just because you asked so nicely.

Added logging to all admin actions.

Logging is using a Database function instead of PublicVariableEventHandler(it can still print to rpt if u have any reason to do so)

Changed the getID button in the admin menu to display the log dialog showing admin and player logs, initial commits actually had this in the commit messages but i removed those when i squashed the commits together just to make it nice and clean for anyone who has to read/implement it. it should now be displayed alongside the rest of the playerinfo within the admin menu.

Is there even anyone using getID??

Player textures converted to use a config file ie.
{ "textures\cop_uniform.jpg", "life_coplevel", "SCALAR", 1, "cop"},

Additional fixes

Typo in localize string, fn_restrain.sqf
Fix to stop vinteraction menu from opening while restrained.
Fix to stop players from entering turrets on vehicles while being restrained.

it's got some merge conflicts after the last few merges that i need to fix later because i dont have time right now.

**Comment by tkcjesse** [comment]
I've already brought up removing GetID completely. Yes people use it. Those who don't use a remote admin tool like DaRT. It's for in-game bans and kicks.

File: life_server/Functions/Systems/fn_logIt.sqf [hunk 7]
```diff
@@ -1,8 +1,39 @@
...
+
+if(_message == "") exitWith {}; 
+
+_timestamp = "extDB2" callExtension "9:LOCAL_TIME";
+_timestamp = [_timestamp] call DB_fnc_mrestoArray;
+
+if(_admin) then {
+
+	if(EQUAL(LIFE_SETTINGS(getNumber,"player_rptlog"),1)) then {
+		diag_log format["[LOG] ADMIN %1", _log];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_logIt.sqf]: _log isn't defined yet

Comment (Liam0) [inline life_server/Functions/Systems/fn_logIt.sqf]: Thanks

**Pull request closed without merging**